sparkle/lib/target/risc-v/encoder/data.cpp

```
@@ -0,0 +1,71 @@
+            /* if out of range, set the offset to maximum */
+            offset = (offset < 0) ? -1048576 : 1048574;
+        }
+        
+        /* extract bits to get the J-type immediate */
+        int imm20 = (offset >> 1) & 0xFFFFF;
+        int imm_20 = (imm20 >> 19) & 0x1;
+        int imm_19_12 = (imm20 >> 11) & 0xFF;
+        int imm_11 = (imm20 >> 10) & 0x1;
+        int imm_10_1 = imm20 & 0x3FF;
+        
+        return (imm_20 << 31) |     /* imm[20] */
```

Cppcheck (2.10-2)

[error] Signed integer overflow for expression 'imm_20<<31'.

(integerOverflow)